fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/GroupByExpression.java

```diff
@@ -471,13 +471,13 @@ private BooleanWithConstraint groupingSubsumedBy(@Nonnull final Quantifier candi
                         }
 
                         return comparisons.stream()
-                                .anyMatch(comparison -> comparison.getType().isEquality() &&
-                                        comparison.getType() == Comparisons.Type.EQUALS);
+                                .anyMatch(comparison -> (comparison.getType().isEquality() &&
+                                        comparison.getType() == Comparisons.Type.EQUALS) || (comparison.getType() == Comparisons.Type.IS_NULL));
```

Comment:
Can you use the comparison types equality test instead of enumerating the types you like: `comparison.getType().isEquality()`

Reply:
I changed it to : `c.isEquality() && (c==EQUALS || c== IS_NULL)`, a couple of `TEXT_CONTAINS` comparison types also returns `isEquality() = true`, so I guess we have to check whether it is EQUALS or IS_NULL.